### PR TITLE
[BUGFIX beta] Functional update for adapter populated record arrays

### PR DESCRIPTION
**Pull request by fivetanley**
This is an updated version of https://github.com/emberjs/data/pull/3086/ by @vinilios.

permit for adapter populated record array contents to be updated by
calling `update` method.

refs #3082

cc @wecc @igorT

### CODE REVIEW

**Comment by fivetanley** [comment]
also @tchak

**Comment by tchak** [comment]
LGTM :+1:

**Comment by wecc** [comment]
:+1:

File: tests/unit/adapter-populated-record-array-test.js [hunk 1]
```diff
@@ -73,3 +73,61 @@ test('recordArray.replace() throws error', function(assert) {
     recordArray.replace();
   }, Error("The result of a server query (on (subclass of DS.Model)) is immutable."), 'throws error');
 });
+
+test("when an adapter populated record gets updated the array contents are also updated", function(assert) {
```
Comment (sly7-7) [inline tests/unit/adapter-populated-record-array-test.js]: maybe missing `array` here: `adapter populated record **array** gets...`

**Pull request merged**